### PR TITLE
:rotating_light: :rotating_light: Source Mixpanel: fix typing

### PR DESCRIPTION
**Pull request by davydov-d**
## What
1) Fix date field: https://github.com/airbytehq/oncall/issues/1772
2) Fix the primary key for the export stream to identify a record uniquely: [#2939](https://github.com/airbytehq/oncall/issues/2939)

## How
1) Remove `"format": "date-time"` from date-time fields as this is not quite correct. Mixpanel's datetime data can be either of
 - "2021-12-16T00:00:00"
 - "1638298874"
 - "15/08/53895"
 2) Add fields to the primary key of export stream to identify a record uniquely

 ## 🚨 User Impact 🚨
 Users will have to refresh their schemas and refresh data as some of their destination columns may change their type from Data/Time/Datetime to string. Primary key for export stream changed. [Breaking changes doc.](https://docs.google.com/document/d/1PVlFSz4vFBtbF1HszPNs4dzz76Nh-nrmwIhNcnywsXw/edit)

### CODE REVIEW

**Review by darynaishchenko — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-mixpanel test report (commit [`f29f0b0c0a`](https://github.com/airbytehq/airbyte/pull/30025/commits/f29f0b0c0a58d76fec05442f6cfbf1633a47d467)) - ❌

⏲️  Total pipeline duration: 68mn40s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Connector package install                                              | ✅        |
| Build source-mixpanel docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |
| Code format checks                                                     | ✅        |
| Validate airbyte-integrations/connectors/source-mixpanel/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ❌        |
| QA checks                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ddavydov_1772-fix-datetime-field-typing-for-mixpanel/1695824556/f29f0b0c0a58d76fec05442f6cfbf1633a47d467/source-mixpanel/0.1.41/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f29f0b0c0a58d76fec05442f6cfbf1633a47d467)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mixpanel test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-mixpanel test report (commit [`0103f3b10a`](https://github.com/airbytehq/airbyte/pull/30025/commits/0103f3b10a2a087c6f1eb0c7694e98750595331d)) - ✅

⏲️  Total pipeline duration: 70mn02s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Connector package install                                              | ✅        |
| Build source-mixpanel docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |
| Code format checks                                                     | ✅        |
| Validate airbyte-integrations/connectors/source-mixpanel/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ddavydov_1772-fix-datetime-field-typing-for-mixpanel/1695829395/0103f3b10a2a087c6f1eb0c7694e98750595331d/source-mixpanel/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:0103f3b10a2a087c6f1eb0c7694e98750595331d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mixpanel test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-mixpanel test report (commit [`aee4755dea`](https://github.com/airbytehq/airbyte/pull/30025/commits/aee4755dea0c99778c9fb67470bcd20ce5a7a32e)) - ✅

⏲️  Total pipeline duration: 66mn54s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Connector package install                                              | ✅        |
| Build source-mixpanel docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |
| Code format checks                                                     | ✅        |
| Validate airbyte-integrations/connectors/source-mixpanel/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ddavydov_1772-fix-datetime-field-typing-for-mixpanel/1695890829/aee4755dea0c99778c9fb67470bcd20ce5a7a32e/source-mixpanel/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:aee4755dea0c99778c9fb67470bcd20ce5a7a32e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mixpanel test
```

**Review by clnoll — COMMENTED** [review]
Looks great, just one small question about the test, @davydov-d.

File: airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py:298]: Yes, the test was rewritten. It is still testing schema creation for the _engage_ stream.

Comment (clnoll) [reply airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py:298]: I see that it's testing schema creation but for the other streams the tests also cover `stream.read_records` and `records_length`. I didn't see a separate test where that would be covered for the engage stream. Can we leave these in, or put them in a new test?

Comment (tolik0) [reply airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py:298]: Previously, EngageSchema.read_records was called, but now Engage.get_json_schema [does the same](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/engage.py#L194). Regarding records_length, we now compare the content of the schemas, not just the number of fields.

Comment (clnoll) [reply airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py:298]: Got it, thank you for the code link!

**Review by clnoll — APPROVED** [review]
LGTM pending the approval of the breaking changes doc!

**Comment by airbyte-oss-build-runner** [comment]
##  source-mixpanel test report (commit [`56ff05661c`](https://github.com/airbytehq/airbyte/pull/30025/commits/56ff05661c6b261b0921da7032d01400f95c4e1c)) - ❌

⏲️  Total pipeline duration: 130mn08s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-mixpanel docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                   | ✅        |
| Acceptance tests                                             | ❌        |
| Code format checks                                           | ✅        |
| Validate metadata for source-mixpanel                        | ✅        |
| Connector version semver check                               | ✅        |
| Connector version increment check                            | ✅        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ddavydov_1772-fix-datetime-field-typing-for-mixpanel/1696030899/56ff05661c6b261b0921da7032d01400f95c4e1c/source-mixpanel/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:56ff05661c6b261b0921da7032d01400f95c4e1c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mixpanel test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-mixpanel test report (commit [`5068013a85`](https://github.com/airbytehq/airbyte/pull/30025/commits/5068013a85d97ec3e1d19fe12c8dd9802198d9d4)) - ❌

⏲️  Total pipeline duration: 65mn09s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-mixpanel docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                   | ✅        |
| Acceptance tests                                             | ❌        |
| Code format checks                                           | ✅        |
| Validate metadata for source-mixpanel                        | ✅        |
| Connector version semver check                               | ✅        |
| Connector version increment check                            | ✅        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ddavydov_1772-fix-datetime-field-typing-for-mixpanel/1696266202/5068013a85d97ec3e1d19fe12c8dd9802198d9d4/source-mixpanel/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5068013a85d97ec3e1d19fe12c8dd9802198d9d4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mixpanel test
```

**Comment by grkhr** [comment]
@tolik0 @davydov-d 

Hey folks! This primary key edit (`for export`) is still non-accurate.

https://github.com/airbytehq/airbyte/blob/5001ec1cca164a7a23f313cdf814a4748415c086/airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/export.py#L78

You are assuming that a user can't send multiple events with the same name in one second, but it's happening every time with high frequency events. Also properties can be different.
MixPanel uses `insert_id` as the unique "event". Please consider using it or just leave that choice to the end user. Now it's impossible to change primary key in UI, it's locked.

**Pull request merged**
